src/ui/components/menu/Options.tsx

```diff
@@ -58,49 +61,68 @@ export default class Options extends React.Component<IProps, IState> {
   async componentDidMount() {
     const config = await getConfig()
     const languages = await getLanguages()
-    
+
     // Remove jar from path
     const path = config.grasscutter_path.replace(/\\/g, '/')
     const folderPath = path.substring(0, path.lastIndexOf('/'))
     const encEnabled = await server.encryptionEnabled(folderPath + '/config.json')
 
     this.setState({
       game_install_path: config.game_install_path || '',
+      game_executable: config.game_executable || '',
       grasscutter_path: config.grasscutter_path || '',
       java_path: config.java_path || '',
       grasscutter_with_game: config.grasscutter_with_game || false,
       language_options: languages,
       current_language: config.language || 'en',
       bg_url_or_path: config.customBackground || '',
-      themes: (await getThemeList()).map(t => t.name),
+      themes: (await getThemeList()).map((t) => t.name),
       theme: config.theme || 'default',
-      encryption: await translate(encEnabled ? 'options.enabled' : 'options.disabled')
+      encryption: await translate(encEnabled ? 'options.enabled' : 'options.disabled'),
     })
 
     this.forceUpdate()
   }
 
-  setGameExec(value: string) {
+  setGamePath(value: string) {
     setConfigOption('game_install_path', value)
 
     this.setState({
-      game_install_path: value
+      game_install_path: value,
     })
   }
 
+  setGameExecutable(value: string) {
```

Comment:
A less vital spot but early returns here would be nice too (and if there are parts of the code where I have not done that dw I will go through and fix em all up to be uniform at some point)